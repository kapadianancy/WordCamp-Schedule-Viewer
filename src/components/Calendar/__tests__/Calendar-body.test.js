/* eslint-disable no-undef */
/**
 * @jest-environment  jsdom
 */
import React from 'react'
import { render, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom'
import CalendarBody from '../Calendar-body'

describe('#Calendar Body', () => {
  const props = {
    themeClass: 'dark'
  }
  const originalFetch = global.fetch

  beforeEach(() => {
    jest.spyOn(window, 'alert').mockImplementation(() => {})
    global.fetch = jest.fn(() => Promise.resolve({
      json: () => Promise.resolve({
        value: 'Testing something!'
      })
    }))
  })

  // Part 3
  afterAll(() => {
    global.fetch = originalFetch
    cleanup()
  })

  it('should render without crashing', () => {
    const { getByTestId } = render(<CalendarBody {...props}/>)
    const component = getByTestId('calendar-body-test')
    expect(component).toBeInTheDocument()
  })

  it('should render theme correctly', () => {
    const component = render(<CalendarBody {...props}/>)
    expect(component.container.querySelector('.card').classList.contains('dark')).toBe(true)
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<CalendarBody {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
