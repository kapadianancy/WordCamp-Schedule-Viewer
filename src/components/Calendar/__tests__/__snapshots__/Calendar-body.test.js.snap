// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Calendar Body should match snapshot 1`] = `
<div
  className="content-wrapper"
  data-testid="calendar-body-test"
  style={
    Object {
      "fontSize": "15px",
      "fontWeight": "bold",
      "marginRight": "2px",
    }
  }
>
  <div
    className="card dark"
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <div
      className="card-body"
      style={
        Object {
          "fontSize": "15px",
          "fontWeight": "bold",
          "marginRight": "2px",
        }
      }
    >
      <div
        className="calendarWrapper"
      >
        <div
          className="modeWrapper"
        >
          <button
            className="modeButton"
            onClick={[Function]}
          >
            Year
          </button>
          <button
            className="modeButton modeButtonActive"
            onClick={[Function]}
          >
            Month
          </button>
          <button
            className="modeButton"
            onClick={[Function]}
          >
            Day
          </button>
        </div>
        <div
          className="calendarHeader"
        >
          <h1>
            <span
              className="thickText"
            >
              July
            </span>
            Â 
            <span
              className="thinText"
            >
              2021
            </span>
          </h1>
          <div
            className="calendarHeaderButtons"
          >
            <button
              onClick={[Function]}
            >
              June
            </button>
            <button
              onClick={[Function]}
            >
              August
            </button>
          </div>
        </div>
        <div>
          <div
            className="calendarDayOfWeek"
          >
            <div>
              Sun
            </div>
            <div>
              Mon
            </div>
            <div>
              Tue
            </div>
            <div>
              Wed
            </div>
            <div>
              Thu
            </div>
            <div>
              Fri
            </div>
            <div>
              Sat
            </div>
          </div>
          <div
            className="weekRow"
          >
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate inactiveDay"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    27
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate inactiveDay"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    28
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate inactiveDay"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    29
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate inactiveDay"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    30
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    1
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    2
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    3
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="weekRow"
          >
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    4
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    5
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    6
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    7
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    8
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    9
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    10
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="weekRow"
          >
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    11
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    12
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    13
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    14
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    15
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    16
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText currentDay"
                >
                  <span>
                    17
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="weekRow"
          >
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    18
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    19
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    20
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    21
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    22
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    23
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    24
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            className="weekRow"
          >
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    25
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    26
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    27
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    28
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    29
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    30
                  </span>
                </div>
              </div>
            </div>
            <div
              className="dayCell"
              id="dayCell"
              onClick={[Function]}
            >
              <div
                className="dayDate"
                id="dayHeader"
              >
                <div
                  className="dayText"
                >
                  <span>
                    31
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
